rlgym_tools/sb3_utils/sb3_log_reward.py

```diff
@@ -1,44 +1,152 @@
+        self.episode_steps += 1
+        return rew
+
+    def get_final_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rew = self.reward_function.get_final_reward(player, state, previous_action)
+        self.reward_sum += rew
+        self.episode_steps += 1
+        return rew
+
+class SB3CombinedLogReward(CombinedReward):
+
+    index_tracker = 0 # So we can uniquely identify each instance
+
+    def __init__(
+            self,
+            reward_functions: Tuple[RewardFunction, ...],
+            reward_weights: Optional[Tuple[float, ...]] = None
+    ):
+        """
+        Creates the combined reward using multiple rewards, and a potential set
+        of weights for each reward. Will also log the weighted rewards to
+        the model's logger if a SB3CombinedLogRewardCallback is provided to the
+        learner.
+
+        :param reward_functions: Each individual reward function.
+        :param reward_weights: The weights for each reward.
+        """
+        super().__init__(reward_functions,reward_weights)
+        self.index = SB3CombinedLogReward.index_tracker
+        SB3CombinedLogReward.index_tracker += 1
+
+        # make sure there is a folder to dump to
+        os.makedirs('bin/combinedlogfiles', exist_ok=True)
+        self.dumpfile = f'bin/combinedlogfiles/{self.index}.txt'
+
+        # clean out any files that are left over from previous runs
+        if self.index == 0:
+            for file in os.listdir('bin/combinedlogfiles'):
+                if '.txt' in file:
+                    os.remove('bin/combinedlogfiles/'+file)
+
+
+        # initiates the array that will store the rewards
+        self.returns = np.zeros(len(self.reward_functions))
+
+    def reset(self, initial_state: GameState):
+        self.returns = np.zeros(len(self.reward_functions))
+        super().reset(initial_state)
+
+    def get_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rewards = [
+            func.get_reward(player, state, previous_action)
+            for func in self.reward_functions
+        ]
+
+        self.returns += [a*b for a, b in zip(rewards, self.reward_weights)] # store the rewards
+
+        return float(np.dot(self.reward_weights, rewards))
+
+    def get_final_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rewards = [
+            func.get_final_reward(player, state, previous_action)
+            for func in self.reward_functions
+        ]
+
+        self.returns += [a*b for a, b in zip(rewards, self.reward_weights)] # store the rewards
+
+        # write the rewards to file and reset
+        with open(self.dumpfile, 'a') as f:
+            # in a list so we can eval to get it back easily
+            f.write('\n' + str(list(self.returns)))
+        self.returns = np.zeros(len(self.reward_functions))
+
+        return float(np.dot(self.reward_weights, rewards))

```

Comment:
Given the significant repetition from `get_reward()`, this should be refactored (potentially so there's just a call to `get_reward()` and an additional call to a `write()`)